tests/pulse/test_parametrized_evolution.py

```diff
@@ -89,6 +89,18 @@ def test_init(self):
         assert qml.math.allequal(ev.data, [1, 2])
         assert qml.math.allequal(ev.parameters, [1, 2])
         assert ev.num_params == 2
+        assert ev.dense is True
+
+    def test_set_dense(self):
+        """Test that flag dense is set correctly"""
+        ops = [qml.PauliX(0), qml.PauliY(1), qml.PauliZ(2)]
+        coeffs = [1, 2, 3]
+        H = ParametrizedHamiltonian(coeffs, ops)
+        ev = ParametrizedEvolution(H=H, params=[1, 2], t=2)
+        assert ev.dense is False
+
+        ev2 = ParametrizedEvolution(H=H, params=[1, 2], t=2, dense=True)
+        assert ev2.dense is True
```

Comment:
Adding tests for less than three wires and `dense=None` as well as for `dense=False`.
```suggestion
        assert ev2.dense is True
        ops = [qml.PauliX(0), qml.PauliY(1)]
        coeffs = [1, 2]
        H = ParametrizedHamiltonian(coeffs, ops)
        ev = ParametrizedEvolution(H=H, params=[1, 2], t=2)
        assert ev.dense is True

        ev2 = ParametrizedEvolution(H=H, params=[1, 2], t=2, dense=False)
        assert ev2.dense is False
```
btw: Where are those `params` going? :o

Reply:
added a test with explicit dense=False, the case of 2 wires is tested in the init, do you want an explicit extra test with 2 wires?

Reply:
Ah no, that's fine :)